fw/access_log.c

```diff
@@ -413,21 +537,93 @@ do_access_log(TfwHttpResp *resp)
 			  TFW_HTTP_RESP_CUT_BODY_SZ(resp));
 }
 
+static int
+cfg_access_log_set(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	int i;
+	const char *val;
+	bool off = false;
+
+	TFW_CFG_CHECK_VAL_N(>, 0, cs, ce);
+	TFW_CFG_CHECK_NO_ATTRS(cs, ce);
+
+	access_log_type = ACCESS_LOG_OFF;
+
+	TFW_CFG_ENTRY_FOR_EACH_VAL(ce, i, val) {
+		if (strcasecmp(val, "off") == 0) {
+			off = true;
+		} else if (strcasecmp(val, "dmesg") == 0) {
+			access_log_type |= ACCESS_LOG_DMESG;
+		} else if (strcasecmp(val, "mmap") == 0) {
+			access_log_type |= ACCESS_LOG_MMAP;
+		} else {
+			T_ERR_NL("invalid access_log value: '%s'\n", val);
+			return -EINVAL;
+		}
+	}
+
+	if (off && access_log_type != ACCESS_LOG_OFF) {
+		T_ERR_NL("access_log 'off' value should be the only value\n");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int
+tfw_access_log_start(void)
+{
+	int cpu;
+
+	tfw_mmap_buffer_free(mmap_buffer);
```

Comment:
This function can called multiple times, on each Tempesta FW reload. On each call `holders_cnt` in mmap_buffer.c
increases and after 4 reload we never can start Tempesta.  There are two options:
- Decrease `holders_cnt` and clear entry in `mmap_buffer.c`. 
- call
```
if (tfw_runstate_is_reconfig())
		return 0;
```